sim_pipeline/gg_lens.py

```diff
@@ -42,32 +42,41 @@ def position_alignment(self):
         return self._center_lens, self._center_source
 
     def validity_test(self, min_image_separation=0, max_image_separation=10):
-        """
-        check whether lensing configuration matches selection and plausibility criteria
-
-        :param min_image_separation:
-        :param max_image_separation:
-        :return: boolean
-        """
-        z_lens = self._lens_dict['z']
-        z_source = self._source_dict['z']
-        if z_lens >= z_source:
-            return False
-        if self._theta_E * 2 < min_image_separation:
-            return False
-        if self._theta_E * 2 > max_image_separation:
-            return False
-        kwargs_model, kwargs_params = self.lenstronomy_kwargs('any_band')
-        lens_model_class = LensModel(lens_model_list=kwargs_model['lens_model_list'])
-        lens_eq_solver = LensEquationSolver(lens_model_class)
-
-        source_pos = kwargs_params['kwargs_source'][0]['center_x'], kwargs_params['kwargs_source'][0]['center_y']
-        # TODO: make image_position definition to not re-compute lens equation solver multiple times
-        image_positions = lens_eq_solver.image_position_from_source(source_pos, kwargs_params['kwargs_lens'])
-        if len(image_positions)<2:    #?????
-            return False
+            """
+            check whether lensing configuration matches selection and plausibility criteria
+
+            :param min_image_separation:
+            :param max_image_separation:
+            :return: boolean
+            """
+            z_lens = self._lens_dict['z']
+            z_source = self._source_dict['z']
+            if z_lens >= z_source:
+                return False
+            if self._theta_E * 2 < min_image_separation:
+                return False
+            if self._theta_E * 2 > max_image_separation:
+                return False
+            kwargs_model, kwargs_params = self.lenstronomy_kwargs('g')
+            lens_model_class = LensModel(lens_model_list=kwargs_model['lens_model_list'])
+            lens_eq_solver = LensEquationSolver(lens_model_class)
+
+            # TODO: make image_position definition to not re-compute lens equation solver multiple times
+            source_pos_x = kwargs_params['kwargs_source'][0]['center_x']
+            source_pos_y = kwargs_params['kwargs_source'][0]['center_y']
+            if (source_pos_x) ** 2 + (source_pos_y) ** 2 > (self._theta_E) ** 2:
```

Comment:
this assumes that the center of the lens is at (0,0). If this is not the case, then this relation does not hold